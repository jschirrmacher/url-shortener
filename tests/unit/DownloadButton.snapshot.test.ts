/**
 * @vitest-environment happy-dom
 */
import { describe, it, expect } from "vitest"
import { mount } from "@vue/test-utils"
import DownloadButton from "~/components/ui/DownloadButton.vue"

describe("DownloadButton", () => {
  it("renders PNG button correctly", () => {
    const wrapper = mount(DownloadButton, {
      props: {
        format: "png",
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it("renders SVG button correctly", () => {
    const wrapper = mount(DownloadButton, {
      props: {
        format: "svg",
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it("renders disabled PNG button correctly", () => {
    const wrapper = mount(DownloadButton, {
      props: {
        format: "png",
        disabled: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it("renders disabled SVG button correctly", () => {
    const wrapper = mount(DownloadButton, {
      props: {
        format: "svg",
        disabled: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it("has correct props for PNG", () => {
    const wrapper = mount(DownloadButton, {
      props: {
        format: "png",
      },
    })

    expect(wrapper.props().format).toBe("png")
  })

  it("has correct props for SVG", () => {
    const wrapper = mount(DownloadButton, {
      props: {
        format: "svg",
      },
    })

    expect(wrapper.props().format).toBe("svg")
  })
})
