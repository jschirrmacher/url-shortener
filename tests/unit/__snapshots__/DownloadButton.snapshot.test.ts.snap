// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DownloadButton > renders PNG button correctly 1`] = `"<base-button-stub variant="secondary" size="sm" disabled="false" title="Als PNG herunterladen"></base-button-stub>"`;

exports[`DownloadButton > renders SVG button correctly 1`] = `"<base-button-stub variant="secondary" size="sm" disabled="false" title="Als SVG herunterladen"></base-button-stub>"`;

exports[`DownloadButton > renders disabled PNG button correctly 1`] = `"<base-button-stub variant="secondary" size="sm" disabled="true" title="Als PNG herunterladen"></base-button-stub>"`;

exports[`DownloadButton > renders disabled SVG button correctly 1`] = `"<base-button-stub variant="secondary" size="sm" disabled="true" title="Als SVG herunterladen"></base-button-stub>"`;
